### DISCUSSION TITLE
Problem with using JUnit4to5Migration

### DISCUSSION THREAD

**Discussion by ErhardSiegl** [openrewrite/rewrite#3911] [Q&A] [answered]
I'm rather new to OpenRewrite. I tried to migrate a project from Junit4 to Junit5 with `org.openrewrite.java.testing.junit5.JUnit4to5Migration`. The Java files worked fine, but the the Jupiter dependencies were not added to the pom.xml. When I changed the recipe in the following way it worked:
```
3c3
< name: org.openrewrite.java.testing.junit5.JUnit4to5Migration
---
> name: org.openrewrite.java.testing.junit5.JUnit4to5Migration1
68c68
<       onlyIfUsing: org.junit.jupiter.api.Test
---
>       onlyIfUsing: org.junit.Test
70a71,79
>       groupId: org.mockito
>       artifactId: mockito-junit-jupiter
>       version: 5.x
>       onlyIfUsing: org.mockito.runners.MockitoJUnitRunner
>       acceptTransitive: true
>   - org.openrewrite.java.dependencies.RemoveDependency:
>       groupId: org.mockito
>       artifactId: mockito-all
>   - org.openrewrite.java.dependencies.AddDependency:
```
It seems that the `onlyIfUsing` needs the original class (`org.junit.Test`) and not the classname which is created by the recipe (`org.junit.jupiter.api.Test`). Is this a bug in the recipe or did I miss anything? I use the following configuration:

```
            <plugin>
                <groupId>org.openrewrite.maven</groupId>
                <artifactId>rewrite-maven-plugin</artifactId>
                <version>5.17.1</version>
                <dependencies>
                    <dependency>
                        <groupId>org.openrewrite.recipe</groupId>
                        <artifactId>rewrite-migrate-java</artifactId>
                        <version>2.5.0</version>
                    </dependency>
                    <dependency>
                        <groupId>org.openrewrite.recipe</groupId>
                        <artifactId>rewrite-testing-frameworks</artifactId>
                        <version>2.1.5</version>
                    </dependency>
                </dependencies>
            </plugin>
```
and start it with `mvn rewrite:run -Drewrite.activeRecipes=org.openrewrite.java.testing.junit5.JUnit4to5Migration1`
[rewrite.yml.txt](https://github.com/openrewrite/rewrite/files/13929460/rewrite.yml.txt)

**Comment by timtebeek** [comment, marked as answer]
hi @ErhardSiegl ! Thanks for the heads up! I've converted your suggestion above into an issue over on https://github.com/openrewrite/rewrite-testing-frameworks/issues/457. Let's discuss the details for a fix there. :)
